### PR TITLE
cg_rocket_datasource: sort resolutions by size (larger first)

### PR DESCRIPTION
**Pull request by illwieckz**
[sort resolutions](https://dl.illwieckz.net/b/unvanquished/bugs/sort-resolutions/unvanquished_2022-03-07_042646_000.png)

I don't think `CG_Rocket_SortResolutionList` is used anyway.

### CODE REVIEW

File: src/cgame/cg_rocket_datasource.cpp [hunk 1]
```diff
@@ -547,6 +569,9 @@ void CG_Rocket_BuildResolutionList( const char* )
 
 	buf[ 0 ] = '\0';
 
+	// Sort resolutions by size by default (larger first).
+	qsort( rocketInfo.data.resolutions, rocketInfo.data.resolutionCount, sizeof( resolution_t ), &ResolutionListCmpBySize );
```
Comment (slipher) [inline src/cgame/cg_rocket_datasource.cpp:573]: Use std::sort. `std::sort(rocketInfo.data.resolutions,rocketInfo.data.resolutions+rocketInfo.data.resolutionCount,[](const resolution_t&a, const resolution_t& b){return a.width*a.height < b.width*b.height;});`

File: src/cgame/cg_rocket_datasource.cpp [hunk 2]
```diff
@@ -569,31 +594,16 @@ void CG_Rocket_BuildResolutionList( const char* )
...
-	if ( a->width > b->width ) return -1;
-
-	if ( b->width > a->width ) return 1;
-
-	if ( a->width == b->width )  return 0;
-
-	return 0; // silence compiler
 }
 
 void CG_Rocket_SortResolutionList( const char*, const char *sortBy )
```
Comment (slipher) [inline src/cgame/cg_rocket_datasource.cpp:599]: CG_Rocket_SortResolutionList could indeed be deleted - it's only useful to have a sort function if the table has more than one column.

File: src/cgame/cg_rocket_datasource.cpp (hunk 1 above)
Comment (illwieckz) [reply src/cgame/cg_rocket_datasource.cpp]: Done (your suggestion listed smaller first, turned it to larger first).

**Comment by illwieckz** [comment]
Did the suggested `std::sort` thing and also deleted `CG_Rocket_SortResolutionList()` and `ResolutionListCmpByWidth()`.

**Comment by slipher** [comment]
LGTM

**Pull request merged**